src/env-inl.h

```diff
@@ -101,6 +101,20 @@ inline void Environment::AsyncHooks::set_enable_callbacks(uint32_t flag) {
   fields_[kEnableCallbacks] = flag;
 }
 
+inline Environment::AsyncCallbackScope::AsyncCallbackScope(Environment* env)
+    : env_(env) {
+  env_->async_callback_cntr_++;
+}
+
+inline Environment::AsyncCallbackScope::~AsyncCallbackScope() {
+  env_->async_callback_cntr_--;
+  CHECK_GE(env_->async_callback_cntr_, 0);
+}
+
+inline bool Environment::AsyncCallbackScope::in_tick() {
```

Comment:
> so that mean the call to MakeCallback in that case can be interrupted by another call to MakeCallback?

Yes it can interrupt an existing call to MakeCallback with another call to MakeCallback.

> OK, do you have a link to where that is done in code?

/cc @indutny I can look for it, but curious if you know where that code is off the top of your head.

> What do you mean by "initiate the JS stack"?

That the call by HTTPParser is the first call into JS. That there is no JS stack beneath it.